libzeth/circuits/notes/note.hpp

```diff
@@ -116,7 +112,7 @@ class output_note_gadget : public note_gadget<FieldT>
         libsnark::protoboard<FieldT> &pb,
         const libsnark::pb_variable<FieldT> &ZERO,
         std::shared_ptr<libsnark::digest_variable<FieldT>> rho,
-        std::shared_ptr<libsnark::digest_variable<FieldT>> commitment,
+        std::shared_ptr<libsnark::pb_variable<FieldT>> cm,
```

Comment:
Shall we just use a const reference to a `pb_variable` and remove the pointer?

Reply:
(This question is also relevant to other files in this PR)

Reply:
I am not sure what are the advantages of using const reference compared to pointers. I need to look into it.
Since this should apply to quite a few files, can I open a separate issue about this?

Reply:
I tend to prefer passing refs as inputs and return values of functions. What's good with refs is that "dereferencing" is done automatically, and by using refs instead of ptrs - among others - makes it impossible to dereference to NULL.

> Since this should apply to quite a few files, can I open a separate issue about this?

As you prefer, if it's easier for you to work on that in a 2nd PR feel free to do so, but yeah, this needs to be changed in many files indeed.

Reply:
Issue #191 created.